docs/index.rst

```diff
@@ -648,9 +648,17 @@ You can customize the instantiator in several ways:
         // can combine the different "modes"
         ->instantiateWith(Instantiator::withoutConstructor()->allowExtra()->alwaysForce())
 
+        // use a "namedConstructor"
+        ->instantiateWith(Instantiator::namedConstructor("methodName"))
+
+        // use a callable
+        ->instantiateWith(Instantiator::use(function(string $title): object {
+            return new Post($title); // ... your own logic
+        }))
+
         // the instantiator is just a callable, you can provide your own
-        ->instantiateWith(function(array $attributes, string $class): object {
-            return new Post(); // ... your own logic
+        ->instantiateWith(function(string $title): object {
+            return new Post($title); // ... your own logic
         })
```

Comment:
I don't know if we should promote this... AFAIK this method won't benefit from the hydration step, so it can lead to buggy behaviors.

I'm even wondering if we should not deprecate this feature, in favor of `Instantiator::use()`... WDYT?

Reply:
This doesn't work I don't think. Unless wrapped in Instantiator::use(), the callable's args will be array/class-string.

But to your comment, kind of thing we need an escape hatch for users to do their own thing completely.

Reply:
yes, you're right, I've rollbacked to the original form